src/lib/Autodiff.hs

```diff
@@ -395,8 +392,9 @@ addTangent x y = case getType x of
 
 pack :: Type -> [Atom] -> Atom
 pack ty elems = case ty of
```

Comment:
It seems like this is only used in a single record-specific rule, so I'd suggest inlining this function there.